students/navdeep/session06/test_mailroom.py

```diff
@@ -0,0 +1,95 @@
+from mailroom_part4 import *
+
+def test_initializeDonorDB():
+	assert initializeDonorDB() == {"Navdeep Gill": [1000], "Nick Garzon": [5,2], "Henry Chipman":[100], 
+	"Lorenzo B": [1000], "Torin Stetina": [200.75]}
+
+def test_initializeOptionsDB():
+	assert initializeOptionsDB() == {'t': askUserForDonor, 's': createFileName, 'r': createReport, 'e': exitMessage}
+
+
+def test_displayMenu():
+	assert displayMenu().startswith("1. Type 't'")
+	assert displayMenu().endswith("the program")
+
+def test_menu():
+	donors = {"Navdeep Gill": [1000], "Nick Garzon": [5,2], "Henry Chipman":[100], "Lorenzo B": [1000], "Torin Stetina": [200.75]}
+	menu_db = {'t': askUserForDonor, 's': writeToFile, 'r': createReport, 'e': exitMessage}
+	assert menu(donors, menu_db) == 't'
+
+def test_askUserForDonor():
+    donors = {"Navdeep Gill": [1000], "Nick Garzon": [5,2], "Henry Chipman":[100], "Lorenzo B": [1000], "Torin Stetina": [200.75]}
+    assert askUserForDonor(donors) == "addNewDonor(donor_dict)"
+
+
+def test_donorExists():
+	donors = {"Navdeep Gill": [1000], "Nick Garzon": [5,2], "Henry Chipman":[100], "Lorenzo B": [1000], "Torin Stetina": [200.75]}
+	assert donorExists(donors, "Navdeep Gill") == True
+	assert donorExists(donors, "Kaleb Smith") == False
+
+def test_showDonorNames():
+	donors = {"John Smith" : [100], "Jane Doe": [200]}
+	assert showDonorNames(donors) == ["John Smith", "Jane Doe"]	
+
```

Comment:
Hmm, maybe a test to make sure a donor has a donations list? (ie the key has a list of values with a length and sum greater than 0.